tools/istio-docker.mk

```diff
@@ -37,6 +37,20 @@ GRAFANA_FILES:=addons/grafana/start.sh \
                addons/grafana/pilot-dashboard.json \
                addons/grafana/import_dashboard.sh
 
+ifeq ($(MACH), ppc64le)
```

Comment:
Offhand I'm thinking that GOARCH is the more appropriate variable to test since we'd care about what platform we're compiling for rather than the platform we're compiling on.

Reply:
@mattdelco from what I understand, this file is not meant for building go binaries. Its for building the docker images. And I guess, you would need a ppc64le platform for building the docker images. So, if we use GOARCH here, it won't make much sense, since, docker build won't proceed if its not able to pull the required images on a different platform. Please correct me if I am wrong.

Reply:
> If docker can't handle cross compiles then I suppose all that really could be done is check that the GOARCH is unset or consistent with MACH (i.e., verify that docker is being asked to bundle go binaries that were compiled for the current/native platform), or just punt on this and leave open the possibility of the uncommon case of cross-compile users shooting themselves in the foot.

Well, it doesn't seem like a good idea to leave it just like that. I'll put a check as per your suggestion, which can be something like:
```
if $GOARCH != $MACH
    echo "Some error msg explaining docker images can't be built for cross platform building..."
endif
```
How does that sound?

Reply:
A check sounds fine, just don't forget to account for the case where GOARCH=amd64 and MACH=x86_64.  I suppose one way to keep the comparison in istio-docker.mk straightforward and simplify the diff in the Makefile is to try something like:

<pre>
MACH ?= $(shell uname -m)
ifeq ($(MACH),x86_64)
MACH := amd64
endif

export GOARCH ?= $(MACH)
</pre>

Reply:
Cool, thanks! Added this.